pkg/apis/ceph.rook.io/v1/storage.go

```diff
@@ -15,6 +15,15 @@ limitations under the License.
 */
 package v1
 
+import "fmt"
+
+const (
+	// Bluestore osd storage
+	BlueStore string = "bluestore"
+	// Bluestore RDR storage
+	BlueStoreRDR string = "bluestore-rdr"
+)
```

Comment:
Good k8s API conventions define a type for string const options with the type name as part of the constant. 

Normally, the options themselves would start with capital letters like "BlueStore" or "BlueStoreRDR" just like the go type, but I tend to believe that if we are following a Ceph typing, it is better to use it verbatim, so "bluestore"/"bluestore-rdr" are good here.

```suggestion
type BackingStoreType string

const (
	// Bluestore osd storage
	BlueStoreBackingStore BackingStoreType = "bluestore"
	// Bluestore RDR storage
	BlueStoreRDRBackingStore BackingStoreType = "bluestore-rdr"
)
```

Reply:
Sorry, I had something backwards on naming. I was using this for reference:
https://github.com/rook/rook/blob/a88740413e0bfcb564606dc4bd24a1b2f18e6c52/pkg/apis/ceph.rook.io/v1/types.go#L484-L503

I think it would be like below following those trends.

```go
type StoreType string

const (
    StoreTypeBlueStore StoreType = "bluestore"

    StoreTypeBlueStoreRDR StoreType = "bluestore-rdr"
)
```

Reply:
using the `StoreType` constant.